Intersect.Server.Core/Entities/Entity.cs

```diff
@@ -3056,7 +3058,26 @@ protected virtual void DropItems(Entity killer, bool sendUpdate = true)
             // Spawn the actual item!
             if (MapController.TryGetInstanceFromMap(MapId, MapInstanceId, out var instance))
             {
-                instance.SpawnItem(X, Y, drop, drop.Quantity, lootOwner, sendUpdate);
+                var itemSource = new EntityItemSource
+                {
+                    EntityType = this.GetEntityType(),
+                    EntityReference = new WeakReference<IEntity>(this)
+                };
+
+                if (this is Player player)
+                {
+                    itemSource.Id = player.Id;
+                }
+                else if (this is Npc npc)
+                {
+                    itemSource.Id = npc.Base.Id;
+                }
+                else if (this is Resource resource)
+                {
+                    itemSource.Id = resource.Base.Id;
+                }
+
+                instance.SpawnItem(itemSource, X, Y, drop, drop.Quantity, lootOwner, sendUpdate);
```

Comment:
This should be refactored out into an abstract method that all of the entity types implement, we should not add in type-specific code like this in an abstract class.